### PR TITLE
API auto-tests Middle level

### PR DESCRIPTION
**Pull request by Anastasia-Ger**

### CODE REVIEW

File: src/test/java/iteration_2_middle/DepositMoneyTest.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            ResponseSpecs.requestReturnedOk())
+            .post(depositRequestMaxLimit)
+            .extract()
+            .as(DepositMoneyResponse.class);
+    softly.assertThat(depositRequestMaxLimit.getBalance()).isEqualTo(depositResponse.getBalance());
+
+
+    // User deposits money over limit 5000, upper boundary value: 5001
+    // Negative test
+    DepositMoneyRequest depositRequestOverLimit = DepositMoneyRequest.builder()
```
Comment (nobugs-me) [inline src/test/java/iteration_2_middle/DepositMoneyTest.java:60]: Тут насколько я вижу ты пытаешься два разных кейса уместить в один (до границы и после границы). Для такого случая нам нужно использовать параметризацию, так как шаги тест кейса не меняются, меняются только данные и ожидаемый результат. 

НО тут вступает другое правило, что мы никогда не параметризуем в один тест негативные и позитивные кейсы, поэтому в твоем случае этот тест должен быть разделен на два: 
1) userCanDepositMoneyWithValidAmount
2) userCanNotDepositMoneyWithInvalidAmount

File: src/test/java/iteration_2_middle/DepositMoneyTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            .id(accountId)
+            .balance(4999)
+            .build();
+
+    new DepositMoneyRequester(RequestSpecs.authAsUser(userRequest.getUsername(), userRequest.getPassword()),
+            ResponseSpecs.requestReturnedOk())
+            .post(depositRequestWithinLimit);
+
+
+    // Get customer accounts
```
Comment (nobugs-me) [inline src/test/java/iteration_2_middle/DepositMoneyTest.java:81]: Второй важный момент: я не вижу проверки баланса после отправки запроса на депозит. 

Должны быть следующие тест кейсы: 
userCanDepositMoneyWithValidAmount
Шаг 1: подготовка данных - создание юзера и аккаунта 
Шаг 2: отправка запроса на депозит на корректную сумму 
Шаг 3: проверка, что баланс изменился на отправленную сумму 

Далее этот кейс можно параметризовать так, что переданная корректная сумма будет являться параметром. Я бы добавила тесты на границу - 4999, 5000, 1 

2) userCanNotDepositMoneyWithInvalidAmount
Шаг 1: подготовка данных - создание юзера и аккаунта 
Шаг 2: отправка запроса на депозит на НЕ корректную сумму 
Шаг 3: проверка, что баланс НЕ изменился на отправленную сумму 

Тут можно сделать параметризацию на значения: 5001, -1

**Review by nobugs-me — CHANGES_REQUESTED** [review]
Привет! 

Спасибо за выполненную работу! 

Я прокомментировала детально один тест, нужно будет его пофиксить и все остальные тесты по аналогии и будет зачет.

**Review by nobugs-me — APPROVED** [review]
Отлично, теперь зачет
